fili-core/src/main/java/com/yahoo/bard/webservice/data/DruidQueryBuilder.java

```diff
@@ -132,7 +133,14 @@ public DruidQueryBuilder(
         // Resolve the table from the the group, the combined dimensions in request, and template time grain
         QueryPlanningConstraint constraint = new QueryPlanningConstraint(request, template);
         ConstrainedTable table = resolver.resolve(group.getPhysicalTables(), constraint).withConstraint(constraint);
-        Filter filter = druidFilterBuilder.buildFilters(request.getApiFilters());
+
+        // combine the filters on the requested logical table with the api query filters
+        ApiFilters apiFilters = ApiFilters.merge(
+                logicalTable.getFilters(),
+                request.getApiFilters()
+        );
```

Comment:
This should probably be append, not merge.

Reply:
What happens if they both have filters on the same dimension? I don't think the right option is ever to remove one of the filters. for example, in the case:

logical table fitler: `dim|id-notin[val1,val2]
request filter: `dim|id-in[val3]`

Those filters are not mutually exclusive. There is at least not an obvious way remove either. I think they need to be merged, but I'm not exactly sure what the merging algorithm should be. Again, I thought handing the naively merged ApiFilters off to the DruidFilterBuilder would handle  the logic for us.